### PR TITLE
enable maven checkstyle

### PR DESCRIPTION
**Pull request by chaokunyang**
## Why are these changes needed?
* Bazel checkstyle is invalid in bazel1.1
* maven check style is convenient and explicit to users
* There are many codes violated check style merged into master
* fix all  existing violations

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by zhijunfu** [comment]
> * There are many codes violated check style merged into master

Should this PR also fix the existing violations?  Otherwise after this is merged, master will fail on checkstyle?

**Comment by chaokunyang** [comment]
> > * There are many codes violated check style merged into master
> 
> Should this PR also fix the existing violations? Otherwise after this is merged, master will fail on checkstyle?

If maven checkstyle should be enabled, I'll fix all violations in a new commit.

**Comment by chaokunyang** [comment]
@zhijunfu fixed all existing violations

File: java/api/pom.xml [hunk 1]
```diff
@@ -43,4 +43,13 @@
...
   </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
```
Comment (raulchen) [inline java/api/pom.xml]: Is it possible to only set this in the root pom.xml file?

Comment (chaokunyang) [reply java/api/pom.xml]: Good catch, just fixed

**Comment by chaokunyang** [comment]
There is a test hang, can anyone take a look?
```
============ [2020-01-19T08:17:36.923716] [TEST START] org.ray.api.test.StressTest.testSubmittingManyTasksToOneActor ============
============ [2020-01-19T08:17:36.923869] [INVOKE METHOD] org.ray.api.test.StressTest.testSubmittingManyTasksToOneActor ============
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

**Review by zhijunfu — APPROVED** [review]

File: streaming/java/pom.xml [hunk 2]
```diff
@@ -149,6 +149,12 @@
                 </plugin>
             </plugins>
         </pluginManagement>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
```
Comment (raulchen) [inline streaming/java/pom.xml]: we need to set config files (`checkstyle.xml` and `checkstyle-suppressions.xml`) here

Comment (chaokunyang) [reply streaming/java/pom.xml]: It's set up in parent pom, in plugin management

Comment (raulchen) [reply streaming/java/pom.xml]: Do you refer to `java/pom.xml`? I don't see that config there either.

Comment (chaokunyang) [reply streaming/java/pom.xml]: Please see pluginManagement in Java/pom.xml streaming/Java/pom.xml. We configured in there

Comment (raulchen) [reply streaming/java/pom.xml]: got it

**Review by raulchen — APPROVED** [review]

**Pull request merged**